crates/sui-node/src/metrics.rs

```diff
@@ -136,6 +154,9 @@ pub fn start_metrics_push_task(config: &sui_config::NodeConfig, registry: Regist
             interval.tick().await;
 
             if let Err(error) = push_metrics(&client, &url, &registry).await {
+                // aggressively recreate our client connection if we hit an error
+                // since our tick interval is only every min, this should not be racey
+                client = client.recreate();
```

Comment:
why not just change this to:
```suggestion
                client = MetricsPushClient::new(config.network_key_pair().copy());
```

Reply:
ah, no real reason i suppose. i'll change it.